### ISSUE TITLE
[Ads.Lite] Not working preview or stable

### ISSUE THREAD

**Issue by jenokizm** [xamarin/GooglePlayServicesComponents#334]
hi, my app target android api 28 with androidX.
when i use 118.1.1-preview01 error:
```
Серьезность	Код	Описание	Проект	Файл	Строка	Состояние подавления
Ошибка		error: ShouldDelayBannerRenderingListenerImplementor is not abstract and does not override abstract method zzb(Runnable) in ShouldDelayBannerRenderingListener
public class ShouldDelayBannerRenderingListenerImplementor	MyApp.Droid	C:\Users\Admin\source\repos\MyApp\MyApp.Droid\obj\Debug\90\android\src\mono\com\google\android\gms\ads\formats\ShouldDelayBannerRenderingListenerImplementor.java	4	
```

when i install 71.1720.1 any error:
```
Серьезность	Код	Описание	Проект	Файл	Строка	Состояние подавления
Ошибка	NU1107	Обнаружен конфликт версий для Xamarin.Android.Support.Core.UI. Установите или укажите прямую ссылку Xamarin.Android.Support.Core.UI 28.0.0.3 на проект MyApp.Droid, чтобы устранить эту проблему. 
 MyApp.Droid -> Xamarin.Essentials 1.5.3.1 -> Xamarin.Android.Support.CustomTabs 28.0.0.3 -> Xamarin.Android.Support.Core.UI (= 28.0.0.3) 
 MyApp.Droid -> Xamarin.GooglePlayServices.Ads.Lite 71.1720.1 -> Xamarin.GooglePlayServices.Basement 71.1620.0 -> Xamarin.Android.Support.v4 28.0.0.1 -> Xamarin.Android.Support.Core.UI (= 28.0.0.1).	MyApp.Droid	C:\Users\Admin\source\repos\MyApp\MyApp.Droid\MyApp.Droid.csproj	1	
```

**Comment by tipa** [comment]
Same problem. Would it be possible to ensure that the bindings are able to compile prior to release them on NuGet?

**Comment by moljac** [comment]
@tipa @jenokizm 
Some errors (mainly ACW errors) cannot be detected until some app is built (then ACW and java tools kick in).
There is `samples/BuildAll` which references all nugets and is supposed to detect some of the issues (in last few months mainly `R8` issues), but it cannot detect everything.

In order to speed up migrating samples (from Android.Support to AndroidX) I extracted samples to my repo where I'll keep issue repro samples and other ported (from java/kotlin) samples for testing.

This is the reason I ALWAYS ask for minimal repro cases. I keep them for regression (and not only) testing. 

The more samples the better - for me and implicitly for you.

If you check this issue - no one has hit/encountered this problem until @jenokizm for last few versions. 

Thanks for your feedback/input.

And if you get the chance - minimal repro sample (in any form) would be highly appreciated.

**Comment by tipa** [comment]
Sure, here's the repro, thanks for looking into it :)
[AdsLiteTest.zip](https://github.com/xamarin/GooglePlayServicesComponents/files/4574129/AdsLiteTest.zip)

**Comment by moljac** [comment]
Hey. Thank you.

Excellent

**Comment by r33software** [comment]
check here 
https://stackoverflow.com/questions/61604669/shoulddelaybannerrenderinglistenerimplementor-xamarin-admob-sdk-118-1-1-building

**Comment by moljac** [comment]
@r33software That is an overkill. Though cool trick.

**Comment by tipa** [comment]
So can we expect some fix that is not an overkill? :)

**Comment by moljac** [comment]
I bet you can.

Nur sag mal: wie schnell, moechtest Du des Fix? (des - a weng Frangisch um meine Deutschsprachkentnisse zu kaschieren.

**Comment by r33software** [comment]
it is not the first time where Nuget Package released with an error and definitely it will last for sometime , with the solution mentioned before (the over kill)  , i totally forgot that the error exist , i already upgraded my app to show adaptive banners and advanced native Admob ads , and it works just fine.

**Comment by moljac** [comment]
@jenokizm @tipa @r33software 

71.1720.2 is out.

https://www.nuget.org/packages/Xamarin.GooglePlayServices.Ads.Lite/71.1720.2

Depends on 

https://www.nuget.org/packages/Xamarin.GooglePlayServices.Basement/71.1620.2

which depends on:

https://www.nuget.org/packages/Xamarin.Android.Support.v4/28.0.0.3

Not 28.0.0.1

I hope this solves your issue for 71.

1xx.yyy previews02 comming out soon (maybe today)

**Comment by moljac** [comment]
@jenokizm 

Error for preview01 packages is completely different bindings issue. Actually both binding sets (71.* and 1xx.*)  are completely different beasts (Android.Support and AndroidX).

Please next time open 2 issues, because I almost closed this issue after 71.xxx updates.

**Comment by jenokizm** [comment]
@moljac yes, I just checked 71.1720.2. it works well, thanks!
And as for version previews02 for AndroidX, we look forward to it.

**Comment by moljac** [comment]
@jenokizm Glad to hear that! Thanks for feedback.

Regarding AndroidX - working on it. I think I might re-bind it from scratch.

BTW - next time open 2 issues for those 2 flavors. Thanks

**Comment by tipa** [comment]
Just installed and tested v118.1.1-preview02, the issue is still present unfortunately.

**Comment by moljac** [comment]
@tipa 

See comment above your last comment

> Regarding AndroidX - working on it. I think I might re-bind it from scratch.

https://github.com/xamarin/GooglePlayServicesComponents/issues/334#issuecomment-627604089

I must rebind that library from scratch.

Did you try old one or do you need 18.1.1  (AndroidX)?

**Comment by tipa** [comment]
Ah ok, I thought a new NuGet package means you already finished those bindings.
I can still use the old one for now, but I would much rather prefer using the AndroidX bindings :)

**Comment by moljac** [comment]
Morgen @tipa 

In `config.json` there are I think 94 packages with v.1xx.yy (AndroidX), slightly less for Android.Support (71.xx) add to that 70 AndroidX and 60 (Android.Support) I maintain and it is easy to understand that sometimes it is easier to do bulk version bump.

This time there was also several improvements for all packages (build tool options like Manfiest Merger and few new targets).

I published to buy some time, hoping other users of Ads.Lite will not hit this issue and they will have better experience with other issues and improvements.

Bottom line: I really work on this with highest priority.

**Comment by tipa** [comment]
Dankeschön :)
Happy to hear an update is in the works - thanks a lot for your efforts!

**Comment by tipa** [comment]
Sorry to bug but have you been able to make any progress on this issue? :)

**Comment by moljac** [comment]
@tipa TL&DR; Yes.

I have added migrated samples (to AndroidX) and saw few more issues similar to that one and I would like to fix those too.

**Comment by jenokizm** [comment]
I wonder what versions you use for binding from https://mvnrepository.com/artifact/com.google.android.gms/play-services-ads-lite?repo=google

**Comment by moljac** [comment]
@tipa Timo. You don't bug me at all.  Believe me I would like to fix this more than you. Seems like I have hit some tooling limitations when generating code. Or bug. Not sure yet. I plan to escalate this.

**Comment by moljac** [comment]
@jenokizm 

there are currently 3 versions of GPS-FB with AndroidX dependencies:

AndroidX branch started in 2019-09 (approximately):

https://github.com/xamarin/GooglePlayServicesComponents/tree/master_based_androidx

Update from 202002

https://github.com/xamarin/GooglePlayServicesComponents/tree/master_based_androidx_based_updates_202002

Update from 202005

https://github.com/xamarin/GooglePlayServicesComponents/tree/master_based_androidx_based_updates_202005

Build is configured in `config.json` which will give you versions:

```
201909      18.1.1      nuget   118.1.1
202002      18.3.0      nuget   118.3.0
202005      19.1.0      nuget   119.1.0
master      17.2.0      nuget   71.1720.2   
```

For AndroidX (Android.Support) and Google.Play.Services-Firebase we use official maven repo:

https://maven.google.com/web/index.html

I use https://mvnrepository.com/ for external dependencies like guava, grpc, opencensuse etc... It differs a bit (somemtimes)

**Comment by jenokizm** [comment]
@moljac okay, it looks fresh, thank you very much for the explanation)

**Comment by moljac** [comment]
https://github.com/xamarin/GooglePlayServicesComponents/commit/7254293415888155e81c9f69584cd89d9988d5ed

**Comment by moljac** [comment]
Related

https://github.com/xamarin/GooglePlayServicesComponents/issues/361

**Comment by tipa** [comment]
From all I can see latest preview version seems to work !  🙂👍  🎉

**Comment by moljac** [comment]
Thanks @tipa!

**Comment by moljac** [comment]
Closing this issue
